src/Microsoft.DotNet.Build.CloudTestTasks/PackageFiles/DumplingHelper.py

```diff
@@ -5,44 +5,74 @@
 import time
 import sys
 import subprocess
+import string
 
 def get_timestamp():
   print(time.time())
 
 def install_dumpling():
   if (not os.path.isfile(dumplingPath)):
     url = "https://dumpling.azurewebsites.net/api/client/dumpling.py"
-    urllib.urlretrieve(url, "dumpling.py")
-    subprocess.call([sys.executable,"dumpling.py", "install", "--update"])
+    scriptPath = os.path.dirname(os.path.realpath(__file__))
+    downloadLocation = scriptPath + "/dumpling.py"
+    urllib.urlretrieve(url, downloadLocation)
+    subprocess.call([sys.executable, downloadLocation, "install", "--update"])
   subprocess.call([sys.executable, dumplingPath, "install", "--full"])
 
+def ensure_installed():
+  if (not os.path.isfile(dumplingPath)):
+    print("Dumpling has not been installed yet. Please run \"DumplingHelper.py install_dumpling\" before collect_dumps.")
+    sys.exit(1)
```

Comment:
+1.  Can log a bunch of stuff but this is not going to end well if something goes wrong for dumpling's installer.